HIRS_Utils/src/main/java/hirs/tpm/eventlog/uefi/UefiSignatureData.java

```diff
@@ -0,0 +1,186 @@
+ */
+public class UefiSignatureData {
+    /** UEFI Certificate GUID.*/
+    private byte[] guid = new byte[UefiConstants.SIZE_16];
+    /** UEFI Signature data.*/
+    private byte[] sigData = null;
+    /** UEFI Certificate object .*/
+    private UefiX509Cert cert = null;
+    /** UEFI Certificate GUID.*/
+    private UefiGuid efiGuid = null;
+    /** UEFI Signature type.*/
+    private UefiGuid signatureType = null;
+    /** UEFI Signature validity.*/
+    private boolean valid = false;
+    /** UEFI Certificate SHA1 hash.*/
+    private byte[] binaryHash = new byte[UefiConstants.SIZE_40];
+    /** UEFI Signature data status.*/
+    private String status = "Signature Data contians a valid Certificate";
+
+/**
+ * UefiSignatureData constructor.
+ * @param inputStream The Signature data.
+ * @param sigType  UEFI defined signature type.
+ * @throws IOException if there's an problem reading the input stream.
+ * @throws CertificateException If there a problem parsing the X509 certificate.
+ * @throws NoSuchAlgorithmException if there's a problem hashing the certificate.
+ */
+UefiSignatureData(final ByteArrayInputStream inputStream, final UefiGuid sigType)
+                        throws IOException, CertificateException, NoSuchAlgorithmException {
+  signatureType = sigType;
+  // UEFI spec section 32.5.3.3 states that SignatureListType of EFI_CERT_SHA256_GUID
+  // only contains a hash, not a cert
+  if (sigType.getVendorTableReference().equals("EFI_CERT_SHA256_GUID")) {
+      inputStream.read(guid);
+      efiGuid = new UefiGuid(guid);
+      // Should be a SHA256 hash of the "binary"
+      inputStream.read(binaryHash);
+  } else if (sigType.getVendorTableReference().equals("EFI_CERT_X509_GUID")) {
+      inputStream.read(guid);
+      efiGuid = new UefiGuid(guid);
+      // Read in Type and Length separately so we calculate the rest of the cert size
+      byte[] certType = new byte[UefiConstants.SIZE_2];
+      inputStream.read(certType);
+      byte[] certLength = new byte[UefiConstants.SIZE_2];
+      inputStream.read(certLength);
+      int cLength = new BigInteger(certLength).intValue() + UefiConstants.SIZE_4;
+      byte[] certData = new byte[cLength];
+      inputStream.read(certData);
+      // put the cert back together
+      byte[] certBlob = new byte[cLength + UefiConstants.SIZE_4];
+      System.arraycopy(certType, 0, certBlob, 0, UefiConstants.SIZE_2);
+      System.arraycopy(certLength, 0, certBlob, UefiConstants.OFFSET_2, UefiConstants.SIZE_2);
+      System.arraycopy(certData, 0, certBlob, UefiConstants.OFFSET_4, cLength);
+      cert = new UefiX509Cert(certBlob);
+      } else if (sigType.isUnknownUUID()) {
+          //status = "Signature List Type has an unknown GUID: " + efiGuid.toString();
+          status = "Signature List Type has an unknown GUID";
+          return;
+      } else {    // else process as a cert (RH SHIM does this)
+          processC509Cert(inputStream);
+          efiGuid = sigType;
```

Comment:
Why is this parameter being stored in two different variables?

Reply:
Changed the names on the byte arrays to make it less confusing.